### PR TITLE
Update PLY to 3.11

### PR DESCRIPTION
**Pull request by saimn**
Ref #7174 for the update process.

### CODE REVIEW

File: astropy/extern/README.rst [hunk 1]
```diff
@@ -7,17 +7,12 @@ party JavaScript libraries used for browser-based features.
...
 
-- ConfigObj_ (Python 2 and 3 versions): This provides the core config file
-  handling for Astropy's configuration system.
+- ConfigObj_: This provides the core config file handling for Astropy's
+  configuration system.
 
 - PLY_: This is a parser generator providing lex/yacc-like tools in Python.
   It is used for Astropy's unit parsing and angle/coordinate string parsing.
 
-- pytest_: This is the test framework used to collect and run Astropy's tests.
```
Comment (mhvk) [inline astropy/extern/README.rst:16]: Nice catch....

**Comment by mhvk** [comment]
p.s. From `CHANGES.rst`, I see that we do document updates of `PLY` under "Other changes". Could add an entry? (for 4.1)

File: CHANGES.rst [hunk 2]
```diff
@@ -399,6 +399,8 @@ Other Changes and Additions
 
 - Increase minimum asdf version to 2.6.0. [#10189]
 
+- The bundled version of PLY was updated to 3.11. [#10257]
```
Comment (bsipocz) [inline CHANGES.rst:402]: update the PR number

**Comment by bsipocz** [comment]
OK, let's merge this, I'm fixing up the changelog anyway before the release.

Thanks @saimn!

**Comment by saimn** [comment]
Sorry for the PR number, I guess someone opened a PR before I pushed my branch :)

**Comment by pllim** [comment]
😬

**Pull request merged**